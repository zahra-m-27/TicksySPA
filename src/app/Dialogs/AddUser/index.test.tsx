import ReactDOM from 'react-dom';
import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import AddUserDialog from './index';

const sample = <AddUserDialog />;

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('render header', () => {
  const {getByText} = render(sample);
  getByText(/افزودن کاربر/i);
});

it('render name label', () => {
  const {getByText} = render(sample);
  getByText(/نام و نام خانوادگی/i);
});

it('render email label', () => {
  const {getByText} = render(sample);
  getByText(/ایمیل/i);
});

it('render role label', () => {
  const {getByText} = render(sample);
  getByText(/عنوان نقش/i);
});

it('render buttons', () => {
  const {getByText} = render(sample);
  getByText(/کاربر جدید/i);
  getByText(/ثبت کاربر/i);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
