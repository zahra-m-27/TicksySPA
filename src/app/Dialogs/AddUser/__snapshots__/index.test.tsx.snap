// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <span>
      افزودن کاربر
    </span>
    <img
      src="add_user_menu.svg"
    />
  </div>
  <div
    className="content"
  >
    <img
      className="avatar"
      src="add_user.svg"
    />
    <div
      className="container"
    >
      <label
        className="label"
      >
        ایمیل
      </label>
      <div
        className="input_container"
      >
        <img
          className="icon email_icon"
          src="search.png"
        />
        <input
          className="input_class"
          dir="auto"
          onChange={[Function]}
          onKeyDown={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="container"
      data-open={false}
    >
      <p
        className="label"
      >
        عنوان نقش
      </p>
      <div
        className="content"
        onClick={[Function]}
      >
        <img
          alt="arrow down"
          className="arrow_down"
          src="arrow_down.svg"
        />
        <p
          className="selected_item_label"
        >
          انتخاب كنيد
        </p>
      </div>
      <div
        className="items"
      >
        <p
          className="item"
          onClick={[Function]}
        >
          نقش 1
        </p>
        <p
          className="item"
          onClick={[Function]}
        >
          نقش 2
        </p>
      </div>
    </div>
    <div
      className="buttons"
    >
      <div
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
      >
        کاربر جدید
      </div>
      <div
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#1354ac",
          }
        }
      >
        ثبت کاربر
      </div>
    </div>
  </div>
</div>
`;
